### ISSUE TITLE
Editing a flag - lose the dependency

### ISSUE THREAD

**Issue by PJNorrisS1** [moloch--/RootTheBox#585]
A Frustrating issue where we are editing flags. On most occasions it will not load the flag dependency, so when we save the flag, we lose the dependency. 9/10 it's repeatable. On occasion it will populate. There is nothing specific... all types of flag and also brand new flags and old flags...

**Comment by eljeffeg** [comment]
I'm not able to repeat that one.  Perhaps when you're able to reproduce it, open the web developer panel and see if there are any errors or something.  I'll need a little more info.

**Comment by PJNorrisS1** [comment]
There is nothing in the docker logs; nothing that appears erroneous in the console developer panel.
I recorded a short screen video of what I'm experiencing. In the video, I'm editing the same flag. There is a dependency on the flag; but you will see - I open and close the same flag, then a couple of times, the dependency appears. There is no pattern!
https://vimeo.com/864657574/df019998db?share=copy

**Comment by bmartin5692** [comment]
I'm seeing the following for a bit of extra detail to add on to PJ's video:
- If I refresh the page, I can reproduce it the first time I click edit, every time. 
- For console logs:
  - When you click Edit, it is posting to `/admin/ajax/objects` and I can see the lock_uuid in the json `lock_uuid: "53d35659-1cc6-47a6-8dee-43c8f39b4597"`
  - (May or may not be related, but - ) Right after this call I see 6 warnings from jquery of `Empty string passed to getElementById()` at `jquery.min.js:2:24747`
- UI Inspector
  - The dependent flag dropdown is empty, but if I inspect the element I can see the the value in the list:
````snip
<select id="edit-flag-lock" name="lock_uuid">
<option value=""></option>
<option value="53d35659-1cc6-47a6-8dee-43c8f39b4597">Welcome</option>
.......more options
````

**Comment by eljeffeg** [comment]
It looks like it could be a javascript issue with `static/js/pages/admin/view/game_objects.js`.  This is the code that should set the option as `selected`.
https://github.com/moloch--/RootTheBox/blob/286d709002b86d3660a6a7a908172f65daa10387/static/js/pages/admin/view/game_objects.js#L47-L52

I still haven't been able to reproduce it here.  What browser are you using?  What I'd suggest is adding a `console.log` after line 5 to print out the (key, value) to the console.  Add a console.log after line 47 to see if it triggers and try to understand what is not happening there.  You could also do this via the browser dev by adding a break point and reviewing the vars but I tend to like seeing it printed out and printing out the logic... If you can track it down there, I'm happy to update.

**Comment by PJNorrisS1** [comment]
This is bizarre. I have added the console logs at line 48

<img width="164" alt="image" src="https://github.com/moloch--/RootTheBox/assets/106666319/c76504b1-5118-4bb9-a7af-c4fe2b89faf0">

Which is correct and value exists. If I step through the script, the value populates.

I have tested on both Chrome and Safari. Logs from Safari.

The entry does not appear in the GUI. This can't be a GUI issue, as if I save the flag entry, then it clears the lock_uuid field and I have to re-edit and find the flag dependency.
